crates/database/src/lib.rs

```diff
@@ -49,42 +50,60 @@ use {
 pub type PgTransaction<'a> = sqlx::Transaction<'a, sqlx::Postgres>;
 
 /// The names of tables we use in the db.
-pub const TABLES: &[&str] = &[
-    "orders",
-    "trades",
-    "invalidations",
-    "last_indexed_blocks",
-    "quotes",
-    "settlements",
-    "presignature_events",
-    "order_quotes",
-    "solver_competitions",
-    "auctions",
-    "competition_auctions",
-    "onchain_placed_orders",
-    "ethflow_orders",
-    "order_execution",
-    "interactions",
-    "ethflow_refunds",
-    "settlement_scores",
-    "settlement_observations",
-    "auction_prices",
-    "auction_participants",
-    "app_data",
-    "jit_orders",
-];
+static TABLES: OnceCell<&'static [&'static str]> = OnceCell::const_new();
 
 /// The names of potentially big volume tables we use in the db.
 pub const LARGE_TABLES: &[&str] = &["order_events"];
 
-pub fn all_tables() -> impl Iterator<Item = &'static str> {
-    TABLES.iter().copied().chain(LARGE_TABLES.iter().copied())
+pub async fn get_table_names(ex: &mut PgConnection) -> sqlx::Result<&'static [&'static str]> {
+    TABLES
+        .get_or_try_init(|| async {
+            #[derive(sqlx::FromRow, Debug)]
+            struct TableName(String);
+
+            let mut query_builder = QueryBuilder::new(
+                "SELECT tablename FROM pg_tables WHERE schemaname = 'public' AND tablename NOT \
+                 LIKE '%flyway%' AND tablename NOT IN (",
+            );
```

Comment:
Without actively removing large tables in this query you can ditch the query builder and simplify / reformat the query string.